autotest/cpp/test_cpl.cpp

```diff
@@ -4301,17 +4301,41 @@ TEST_F(test_cpl, config_file_ignore_env_vars)
     // Reload configuration file
     CPLLoadConfigOptionsFromFile("/vsimem/.gdal/gdalrc", false);
 
-    // Check that environment variables are read as configuration options
-    // and override configuration options
+    // Check that environment variables are read as configuration options and override
+    // configuration options defined in a file. This behavior differs from configuration
+    // options defined outside of a file, in which case the configuration option overrides
+    // the environment variable.
```

Comment:
I've spent some minutes trying to understand both my own original comment and your update proposal, being confused before realising that this was due to the false argument at line 4302.

I guess it would be perhaps clearer to remove that comment, but amend the one at line 4301 by underlying that we pass bOverrideEnvVars = false, and thus configuration options defined in the config file will not override pre-existing environment variables.

Reply:
I also missed that, thanks for pointing it out. I pulled `bOverrideEnvVars` out into a local variable, which should make it more clear what's happening (at least, it does for me)